example/assets/js/components/TourGuide/TourGuide.tsx

```diff
@@ -0,0 +1,88 @@
+import "@mantine/core/styles.css";
+import { Container, TextInput, Button } from "@mantine/core";
+import { useEffect, useState } from "react";
+import classes from "./TourGuide.module.css";
+
+export function TourGuide() {
+  const [latitude, setLatitude] = useState("");
+  const [longitude, setLongitude] = useState("");
+  const [attractions, setAttractions] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    navigator.geolocation.getCurrentPosition(
+      (position: any) => {
+        setLatitude(position.coords.latitude);
+        setLongitude(position.coords.longitude);
+      },
+      (error) => console.log(error)
+    );
+  }, []);
+
+  function findAttractions() {
+    if (!latitude || !longitude) {
+      return;
+    }
+
+    setLoading(true);
+    fetch(`/tour-guide/?coordinate=${latitude},${longitude}`)
+      .then((response) => response.json())
+      .then((data: any) => {
+        console.log(data);
+
+        setAttractions(data.nearby_attractions);
+      })
+      .finally(() => setLoading(false));
+  }
+
+  return (
+    <Container>
+      <div className={classes.searchBar}>
+        <span className={classes.inputBlock}>
+          Latitude:
+          <TextInput
+            value={latitude}
+            onChange={(e) => setLatitude(e.target.value)}
+            className={classes.coordinateInput}
+          />
+        </span>
+        <span className={classes.inputBlock}>
+          Longitude:
+          <TextInput
+            value={longitude}
+            onChange={(e) => setLongitude(e.target.value)}
+            className={classes.coordinateInput}
+          />
+        </span>
+        <Button className={classes.inputBlock} onClick={findAttractions}>
+          Guide Me!
+        </Button>
+      </div>
+      {loading ? <h3>Loading</h3> : null}
```

Comment:
We could still make more use of Mantine components in this example so it would look more similar to the other examples.

Something like (also, `classes` wouldn't be necessary anymore):

```suggestion
    <Container p="md">
      <LoadingOverlay visible={loading} />
      <Group justify="center" align="flex-end">
        <TextInput
          required
          label="Latitude"
          value={latitude}
          onChange={(e) => setLatitude(e.target.value)}
        />
        <TextInput
          required
          label="Longitude"
          value={longitude}
          onChange={(e) => setLongitude(e.target.value)}
        />
        <Button onClick={findAttractions}>Guide Me!</Button>
      </Group>
```

Reply:
Excelent, thanks!